### PR TITLE
Scorched Earth

### PR DESCRIPTION
**Pull request by p-avital**
To provide a more legible API to Python users, as well as follow the new "expose the minimal surface through callback based bindings, and use the language to re-expose more language specific features" binding philosophy, I went from a blank slate to remake the Python API.

This is also intended as a way to pull back on some of the over-extension we've been doing by attempting to have `py:asyncio` drive `rs:async-std`'s event loop, which is the likely cause of some deadlock issues we were starting to get on `api-changes`.

This PR contains the new API, with the examples fixed to use it.
Documentation is still somewhat lacking.

### CODE REVIEW

File: zenoh/closures.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+	and provides a function that will await the closing of the callback before returning said list.
+	"""
+	def __init__(self, timeout=None):
+		self._vec_ = []
+		self._cv_ = Condition()
+		self._done_ = False
+		self.timeout = timeout
+	
+	@property
+	def closure(self):
```
Comment (kydos) [inline zenoh/closures.py:118]: Neat

**Review by kydos — APPROVED** [review]
Looks good to be merged. Looking forward to give it a try.

File: examples/z_sub_queued.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    args.config) if args.config is not None else zenoh.Config()
+if args.mode is not None:
+    conf.insert_json5(zenoh.config.MODE_KEY, json.dumps(args.mode))
+if args.connect is not None:
+    conf.insert_json5(zenoh.config.CONNECT_KEY, json.dumps(args.connect))
+if args.listen is not None:
+    conf.insert_json5(zenoh.config.LISTEN_KEY, json.dumps(args.listen))
+key = args.key
+
+# zenoh-net code  --- --- --- --- --- --- --- --- --- --- ---
```
Comment (Mallets) [inline examples/z_sub_queued.py:62]: zenoh-net is going to disappearin v0.6.beta1. I'm just leaving the comment as a reminder to do a find and replace in the code once merged.

**Comment by Mallets** [comment]
It looks good also on my side. It aligns pretty well with the look-and-fill with the other binding, modulo language-dependent aspects of course. Green light for merging it.

**Pull request merged**